File: LottieSample/src/androidTest/java/com/airbnb/lottie/samples/LottieAnimationViewTest.java
package com.airbnb.lottie.samples;

import android.app.Activity;
import android.os.Build;
import android.view.LayoutInflater;

import org.junit.Before;
import org.junit.Ignore;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

@RunWith(RobolectricTestRunner.class)
@Config(sdk = Build.VERSION_CODES.P)
public class LottieAnimationViewTest {

  private Activity activity;

  @Before
  public void setup(){
    activity = Robolectric.buildActivity(Activity.class).create().get();
  }

  @Test
  public void inflateShouldNotCrash() {
    LayoutInflater.from(activity).inflate(R.layout.lottie_activity_main, null);
  }
}

File: LottieSample/src/androidTest/java/com/airbnb/lottie/samples/LottieAnimationViewTest.kt
package com.airbnb.lottie.samples

import androidx.fragment.app.Fragment
import androidx.fragment.app.testing.launchFragmentInContainer
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@LargeTest
class LottieAnimationViewTest {
    @Test
    fun inflateShouldNotCrash() {
        class TestFragment : Fragment(R.layout.lottie_activity_main)
        launchFragmentInContainer<TestFragment>()
        onView(withId(R.id.animation_view)).check(matches(isDisplayed()))
    }
}
